Cython/Compiler/ExprNodes.py

```diff
@@ -6499,6 +6587,43 @@ def map_to_simple_call_node(self):
 
     def generate_result_code(self, code):
         if self.type.is_error: return
+        if self.fastcallable_with_types:
```

Comment:
This is a completely separate code path that seems worth being extracted into a new method.